app-modules/portal/src/Notifications/AuthenticatePortalNotification.php

```diff
@@ -63,9 +63,12 @@ public function toEmail(object $notifiable): MailMessage
 
     public function identifyRecipient(): array
     {
+        /** @var ?Contact $contact */
+        $contact = $this->authentication->educatable;
+
         return [
-            $this->authentication->educatable->getKey(),
-            $this->authentication->educatable->getMorphClass(),
+            ! empty($contact) ? $contact->getKey() : null,
+            ! empty($contact) ? $this->authentication->educatable->getMorphClass() : 'anonymous',
         ];
```

Comment:
How about refactoring this into an early return for `if (! $contact)` and return a different array instead of using two ternaries